controllers/userController.js

```diff
@@ -1,36 +1,41 @@
 const User = require('../models/User');
 const { StatusCodes } = require('http-status-codes');
 const CustomError = require('../errors');
-const { createTokenUser, attachCookiesToResponse, checkPermissions } = require('../utils');
+const { createTokenUser, attachCookiesToResponse, checkPermissions } = require(
+  '../utils');
 
 // only admins should be able to see all users + we remove password from the response
+// AKOSREVIEW + we only return non-admin users
 const getAllUsers = async (req, res) => {
   console.log(req.user); //we pass to the browser object with name, userId ad role - check authentication.js line 13
-  const users = await User.find({ role: 'user' }).select('-password');
+  // AKOSREVIEW we can probaby also remove __v
+  const users = await User.find({ role: 'user' }).select('-password -__v');
   res.status(StatusCodes.OK).json({ users });
 };
 
-
 const getSingleUser = async (req, res) => {
-    const user = await User.findOne({ _id: req.params.id }).select('-password');
-    if (!user) {
-      throw new CustomError.NotFoundError(`No user with id : ${req.params.id}`);
-    }
-    checkPermissions(req.user, user._id) //we pass user object that is in the request and id property  in a database of the user we are requesting - see line 15
-    res.status(StatusCodes.OK).json({ user });
-  };
+  const user = await User.findOne({ _id: req.params.id }).select('-password');
+  if (!user) {
+    throw new CustomError.NotFoundError(`No user with id : ${req.params.id}`);
+  }
 
+  checkPermissions(req.user, user._id); //we pass user object that is in the request and id property  in a database of the user we are requesting - see line 15
+  res.status(StatusCodes.OK).json({ user });
+};
 
 const showCurrentUser = async (req, res) => {
   //set up user objest and set it eql to req.user
   //here we dont query database, we are getting user we placed in token
+  // AKOSREVIEW maybe don't return expiry?
   res.status(StatusCodes.OK).json({ user: req.user });
 };
 
 //option 1 -update user with user.save()
 
 const updateUser = async (req, res) => {
   const { email, name } = req.body; // we are looking for name, email in req.body
+
+  // AKOSREVIEW is it intentional this only fails if both are missing?
   if (!email || !name) {
```

Comment:
I was a bit surprised that it doesn't let me update just a single field. Like if I just want to update my user's name. I think we'd rather only fail if _both_ fields are missing (eg: `!email && !name`) ?

Reply:
@akosasante I've tried in postman -I think  ow works correctly. thanks!!
![image](https://github.com/NatachaKey/ecommerce/assets/103755090/d63de854-e75d-48fc-8c3b-8838e9f57c11)

Reply:
Looks good :+1: